app/src/main/kotlin/xyz/duckee/android/MainActivity.kt

```diff
@@ -118,51 +115,55 @@ class MainActivity : ComponentActivity() {
-                            }
-                        }
-
-                        path.startsWith("/detail/") -> {
-                            val detailPattern = "/detail/([\\w-]+)".toRegex()
-                            val matchResult = detailPattern.find(path)
-                            matchResult?.let { result ->
-                                val detailId = result.groupValues[1]
-                                navigationController?.navigateToDetailScreen(detailId)
-                            }
-                        }
-
-                        path == "/explore" -> {
-                            navigationController?.navigateToExploreTab()
-                        }
-
-                        path == "/collection" -> {
-                            navigationController?.navigateToCollectionTab()
-                        }
-
-                        path == "/signin" -> {
-                            navigationController?.navigateToSignInScreen()
-                        }
-
-                        else -> {
-                            Timber.tag("[DuckeeMainActivity]").w("Unhandled deep link path: $path")
-                        }
-                    }
+    private fun handleAirbridgeDeeplink() {
+        Airbridge.handleDeeplink(intent) { deeplink ->
+            handleDeeplink(deeplink)
+        }
+
+        Airbridge.handleDeferredDeeplink { deeplink ->
+            deeplink?.let { handleDeeplink(it) }
+        }
+    }
+
+    private fun handleDeeplink(deeplink: Uri) {
+        val path = deeplink.path ?: return
+        when {
+            path.startsWith("/recipe/") -> {
+                val recipePattern = "/recipe/(\\d+)".toRegex()
+                val matchResult = recipePattern.find(path)
+                matchResult?.let { result ->
+                    val recipeId = result.groupValues[1]
+                    navigationController?.navigateToRecipeScreen(recipeId)
+                }
+            }
+
+            path.startsWith("/detail/") -> {
+                val detailPattern = "/detail/([\\w-]+)".toRegex()
+                val matchResult = detailPattern.find(path)
+                matchResult?.let { result ->
+                    val detailId = result.groupValues[1]
+                    navigationController?.navigateToDetailScreen(detailId)
                 }
             }
-            .addOnFailureListener(this) { e ->
-                Timber.tag("[DuckeeMainActivity]").w(e, "getDynamicLink:onFailure")
+
+            path == "/explore" -> {
+                navigationController?.navigateToExploreTab()
             }
+
+            path == "/collection" -> {
+                navigationController?.navigateToCollectionTab()
+            }
+
+            path == "/signin" -> {
+                navigationController?.navigateToSignInScreen()
+            }
+
+            else -> {
+                Timber.tag("[DuckeeMainActivity]").w("Unhandled deep link path: $path")
+            }
+        }
     }
```

Comment:
Refactored the deeplink handling logic to support both Airbridge deeplinks and deferred deeplinks. This change ensures that the app can handle various types of deeplinks, including those that might be triggered after the app is installed.